Remotion/Validation/Core/Results/ValidationFailure.cs

```diff
@@ -33,18 +35,45 @@ public abstract class ValidationFailure
     [NotNull]
     public string LocalizedValidationMessage { get; }
 
-    protected ValidationFailure (
+    [NotNull]
+    public IReadOnlyCollection<ValidatedProperty> ValidatedProperties { get; }
+
+    public ValidationFailure (
         [NotNull] object validatedObject,
         [NotNull] string errorMessage,
-        [NotNull] string localizedValidationMessage)
+        [NotNull] string localizedValidationMessage,
+        [NotNull] IReadOnlyCollection<ValidatedProperty> validatedProperties)
     {
       ArgumentUtility.CheckNotNull("validatedObject", validatedObject);
+      ArgumentUtility.CheckNotNull("validatedProperties", validatedProperties);
       ArgumentUtility.CheckNotNullOrEmpty("errorMessage", errorMessage);
       ArgumentUtility.CheckNotNullOrEmpty("localizedValidationMessage", localizedValidationMessage);
 
+
       ValidatedObject = validatedObject;
       ErrorMessage = errorMessage;
       LocalizedValidationMessage = localizedValidationMessage;
+      ValidatedProperties = validatedProperties;
+    }
+
+    public static ValidationFailure CreatePropertyValidationFailure (
+        [NotNull] object validatedObject,
+        [NotNull] IPropertyInformation validatedProperty,
+        [CanBeNull] object? validatedPropertyValue,
+        [NotNull] string errorMessage,
+        [NotNull] string localizedValidationMessage)
+    {
+      return new ValidationFailure(validatedObject, errorMessage, localizedValidationMessage,
+          new[] { new ValidatedProperty(validatedObject, validatedProperty, validatedPropertyValue) });
+    }
+
+    public static ValidationFailure CreateObjectValidationFailure (
+        [NotNull] object validatedObject,
+        [NotNull] IReadOnlyCollection<ValidatedProperty> validatedProperties,
```

Comment:
I am not sure why we want to do it like that?
I thought that it would be cleaner especially symmetrical, if we have the factory methods for both and pass it to the private ctor. This way we don't have to call the ctor directly.

Reply:
Hmm...you made the ctor private. I.c. Hmm...ok. then please make it protected (just in case / the type is not sealed) and we need both overloads as factory methods. To me, the property validation failure was just a special case, same as the object failure without properties. I can live with both, but then we need two factories for ObjectFailure.